app/controllers/application_controller.rb

```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   include Authentication
+
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+
+  private
+
+  def record_not_found
+    render file: Rails.public_path.join("404.html").to_s, status: :not_found, layout: false
+  end
```

Comment:
I don't think we need to explicitly render the 404. Rails does it by default in production. Check these two links: https://stackoverflow.com/questions/2385799/how-to-redirect-to-a-404-in-rails 
https://www.writesoftwarewell.com/how-to-return-404-not-found-rails/#:~:text=The%20good%20news%20is%2C%20you,will%20send%20the%20public%2F404.